io/io/src/TBufferMergerFile.cxx

```diff
@@ -29,7 +29,24 @@ TBufferMergerFile::~TBufferMergerFile()
 
 Int_t TBufferMergerFile::Write(const char *name, Int_t opt, Int_t bufsize)
 {
+   // Make sure the compression of the basket is done in the unlocked thread and
+   // not in the locked section.
+   if (!fMerger.GetNotrees())
+      TMemFile::Write(name, opt | TObject::kOnlyPrepStep, bufsize);
+
+   // Instead of Writing the TTree, doing a memcpy, Pushing to the queue
+   // then Reading and then deleting, let's see if we can just merge using
+   // the live TTree.
+   if (fMerger.TryMerge(this)) {
+      ResetAfterMerge(0);
+      return 0;
```

Comment:
> I think that if you want to treat TTrees in a special way, this is not where the special treatment code should go

It is useful for both Histogram and TTree (it avoids the stream-compress-uncompress-unstream cycle and instead use directly the live object), since both are the bread and butter of the ROOT File I would venture that the version that does not support them (or support them slow) would be unused.

> Besides, you are returning zero here, when you should be returning the number of bytes written.

I have to re-review that.  In some cases the number of byte written might be zero even though it was succesful; for example if the file contains only histograms, the input data would be merged into the output histogram but it would not be written to the file.